File: 30-03-18/java/02-Xsquare-and-Chocolate-bars.java
import java.io.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        
        while(t-- > 0)
        {
            String str = input.readLine();
            int n = str.length();
            int dp[] = new int[n+1];
            int maxi = 0;
            for(int i = 2; i < n; i++)
            {
                if( str.charAt(i) == str.charAt(i-1) &&  str.charAt(i) == str.charAt(i-2) )
                    dp[i] = 0;
                else
                    dp[i] = (i-3>0? dp[i-3] : 0) +1;
                
                if(dp[i] > maxi)
                    maxi = dp[i];
            }
            maxi *= 3;
            System.out.println(n - maxi);
        }
        
    }
}

File: Mar-18/30-03-18/python/02-Xsquare-and-Chocolate-bars.py
for _ in range( int(input()) ):
    data = input()
    n = len(data)
    dp = {}
    dp[0] = 0
    dp[1] = 0
    
    maxi = 0
    for i in range(2, n):
        dp[i] = 0
        if(data[i] == data[i-1] and data[i] == data[i-2]):
            dp[i] = 0
        else:
            if(i-3 > 0):
                dp[i] = dp[i-3] +1
            else:
                dp[i] = 1
        
        if(dp[i] > maxi):
            maxi = dp[i]
    maxi *= 3
    print(n - maxi)
